cypress/integration/pages/liveRadio/testsForCanonicalOnly.js

```diff
@@ -44,6 +50,49 @@ export const testsThatFollowSmokeTestConfigForCanonicalOnly = ({
         });
       }
     });
+
+    describe('Radio Schedule', () => {
+      it('should be displayed if there is enough schedule data', () => {
+        const isRadioScheduleOnPage = serviceHasRadioSchedule({
+          service,
+          variant,
+        });
+        cy.log(
+          `Live Radio Page configured for Radio Schedule? ${isRadioScheduleOnPage}`,
+        );
+
+        if (isRadioScheduleOnPage) {
+          // Override is being added twice causing a false negative on the call to the schedule! I think other tests need the override though. Will look at this again
```

Comment:
This is because the test URLs have the override added onto them. If I remove them, the page the test visits is not without live data. So this test removes the extra override, which has been added when getting the url from liveRadioUrl.href in the method getRadioScheduleEndpoint in helper.js file. Instead perhaps the liveRadioUrl.href could be changed, rather than changing it in the test file after using the value returned from the helper method.

Reply:
Ah gotya cheers for pointing to that helper. The value is currently being changed before it is passed into the helper. 

So the flow at the moment is:

1. Replace '?renderer_env=live' with '' in `Cypress.env('currentPath')`
2. Pass that url into the `getRadioScheduleEndpoint` helper
3. The helper replaces 'liveradio' with 'schedule.json'
4. If it's in the test environment, it adds '?renderer_env=live' back to the url.

Is this a way to handle test urls that are passed into Cypress with **or** without the `?renderer_env=live` query?

If you **have** to pass urls with `?renderer_env=live` attached to them in the Cypress settings file anyway, then I'm not sure why the query string is removed, just to then be re-added.

Reply:
What the URL in the config is first used for gives more context to the flow: 
The override is added on the config URL as when the page is first visited (done separately in a beforeEach method before each test), it needs the override on the URL to have the live data which allows you to view the schedule. The following test is then done with that path. THEN in the test it is removed before the path is used to access the schedule as it ends up doubled up when it is added a second time to access the schedule, which causes the test to fail. 
If I remove the part in the helper that adds the override again, the tests fail because the override is then not present in the 'liveRadioUrl.href'. 
So maybe this is all because the page is first accessed using one method (using the url in the config), and then the schedule url is accessed in a different way: with the liveRadioUrl.href, which always has another override 'set' into it, as the condition !liveRadioUrl.query.renderer_env is true even if the URL in the config has the override in it. Maybe this is because if the config url already has the override, it is contained in the string in 'liveRadioUrl', not in the liveRadioUrl.query.renderer_env part. The 'baked in' override in the url set in the config is invisible to the !liveRadioUrl.query.renderer_env condition, so within the if clause the code always adds another one.

Does that make sense 😕

Reply:
I'm convinced that the only thing needed to turn one of the incoming page urls from settings

i.e.
> `'/afrique/bbc_afrique_radio/liveradio?renderer_env=live'` (test)
or
> `'/afrique/bbc_afrique_radio/liveradio'` (live)

into a suitable url for the schedule call (`cy.request(getRadioScheduleEndpoint(liveRadioPath))`)

i.e.
> `'/afrique/bbc_afrique_radio/schedule.json?renderer_env=live'`
or
> `'/afrique/bbc_afrique_radio/schedule.json'`

is: `return liveRadioPath.replace('liveradio', 'schedule.json');`

I think all that other stuff in the helper it to make it more robust and usable elsewhere. But IMO it's total overkill, we only need one line, `return liveRadioPath.replace('liveradio', 'schedule.json');`